### PR TITLE
Review step redesign

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves

Resolves partially #864

## How this PR fixes it
Implements the create Safe flow "Review" step according to [Figma](https://www.figma.com/file/ptTs6lDBeUuLNySroJ5PiF/Web-Master-File?node-id=192%3A8146) 

## How to test it
Navigate to `/demo` and observe the review step.
The received creation values will be fully functional once https://github.com/safe-global/web-core/issues/931 is resolved.

## Screenshots
Screenshot 2022-10-27 at 15 30 38

### CODE REVIEW

File: src/components/new-safe/CreateSafe/index.tsx [hunk 1]
```diff
@@ -39,9 +40,10 @@ const CreateSafe = () => {
       </Grid>
 
       <Grid item xs={1} />
-      <Grid item xs={6}>
+      <Grid item xs={6} sx={{ display: 'flex', flexDirection: 'column', gap: '16px' }}>
```
Comment (katspaugh) [inline src/components/new-safe/CreateSafe/index.tsx:43]: Is this temporary?

File: src/components/new-safe/steps/Step3/index.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+              <Grid item>
+                <Grid container spacing={3}>
+                  <ReviewRow name="Network" value={<ChainIndicator chainId={chain?.chainId} inline />} />
+                  <ReviewRow name="Name" value={<Typography>{newSafeName}</Typography>} />
+                  <ReviewRow
+                    name="Owners"
+                    value={
+                      <Box className={css.ownersArray}>
+                        {safeOwners.map((owner, index) => (
+                          <EthHashInfo address={owner.address} showPrefix={false} key={index} />
```
Comment (katspaugh) [inline src/components/new-safe/steps/Step3/index.tsx:109]: In the figma design, these addresses aren'r shortened.

File: src/components/new-safe/steps/Step3/index.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                          </b>
+                        </Typography>
+                      }
+                    />
+                    <ReviewRow
+                      name=""
+                      value={
+                        <Typography color="text.secondary">
+                          You will have to confirm a transaction with your currently connected wallet.
+                        </Typography>
```
Comment (katspaugh) [inline src/components/new-safe/steps/Step3/index.tsx:144]: This should look like this:
Screenshot 2022-10-27 at 15 48 39

File: src/components/new-safe/CreateSafe/index.tsx (hunk 1 above)
Comment (DiogoSoaress) [reply src/components/new-safe/CreateSafe/index.tsx:43]: Yes

File: src/components/new-safe/steps/Step3/index.tsx [hunk 4]
```diff
@@ -83,92 +67,90 @@ const CreateSafeStep3 = (): ReactElement => {
...
 
-  const { handleSubmit } = formMethods
+  const { handleSubmit, watch } = formMethods
 
-  const onSubmit = (data: CreateSafeStep3Form) => {
-    console.log(data)
+  const allFormData = watch()
+
+  const handleBack = () => {
+    onBack(allFormData)
```
Comment (iamacook) [inline src/components/new-safe/steps/Step3/index.tsx:81]: I would recomment calling `getValues` inside `handleBack`. `watch` causes rerenders.

File: src/components/new-safe/steps/Step3/index.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+          <Grid item>
+            <Grid container spacing={3}>
+              <ReviewRow name="Network" value={<ChainIndicator chainId={chain?.chainId} inline />} />
+              <ReviewRow name="Name" value={<Typography>{data.name}</Typography>} />
+              <ReviewRow
+                name="Owners"
+                value={
+                  <Box className={css.ownersArray}>
+                    {data.owners.map((owner, index) => (
+                      <EthHashInfo address={owner.address} shortAddress={false} showPrefix={false} key={index} />
```
Comment (usame-algan) [inline src/components/new-safe/steps/Step3/index.tsx:97]: This is missing the owner name/ens. I suggest to copy it from the other review component.

File: src/components/new-safe/steps/Step3/index.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+const ReviewRow = ({ name, value }: { name: string; value: ReactElement }) => {
+  return (
+    <>
+      <Grid item xs={3}>
+        <Typography>{name}</Typography>
+      </Grid>
+      <Grid item xs={9}>
+        {value}
+      </Grid>
+    </>
```
Comment (usame-algan) [inline src/components/new-safe/steps/Step3/index.tsx:39]: Please check the font-sizes with the design. Labels are supposed to be 14px.

File: src/components/new-safe/steps/Step3/index.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+                  }
+                />
+                <ReviewRow
+                  name=""
+                  value={
+                    <Typography color="text.secondary">
+                      You will have to confirm a transaction with your currently connected wallet.
+                    </Typography>
+                  }
+                />
```
Comment (usame-algan) [inline src/components/new-safe/steps/Step3/index.tsx:136]: The top spacing of this row is different than in the design. I suggest setting it to 8px.

File: src/components/new-safe/steps/Step3/index.tsx (hunk 4 above)
Comment (DiogoSoaress) [reply src/components/new-safe/steps/Step3/index.tsx:81]: True! good suggestion also because I don't expect these values to change in the review step.

**Review by usame-algan — APPROVED** [review]
👍

**Pull request merged**